include/raft.h

```diff
@@ -818,8 +819,9 @@ struct raft
     unsigned max_catch_up_rounds;
     unsigned max_catch_up_round_duration;
 
+    raft_flags feature_flags;
```

Comment:
Do we need to store this value at all in `struct raft`? I'd expect this to be a compile-time constant, in other word `struct raft` will always set `raft_append_entries_result->flags` to whatever the capabilities were at the time the code was compiled. I wouldn't expect `raft->feature_flags` to store a different value than that, unless I'm missing some detail of the mechanics.

Reply:
After reading the reply to the `RAFT_FLAGS_FLAGS` comment, I assume the use case for this field is to tweak it in tests to exercise combinations of legacy and newer nodes?

Reply:
I can imagine it not being a compile-time constant, maybe some option is set at initialization and it changes the flags.

Reply:
> I can imagine it not being a compile-time constant, maybe some option is set at initialization and it changes the flags.

If that's the only reason we have for now, what do you think of making it a compile-time constant for now and turn it into a field only once we'll actually have a use case for dynamically configured features?

Note that in general for supporting dynamically configured features we might need specific logic (if/else), while if we stick to a compile-time constant `struct raft` can assume that all features are on locally, and only have logic to handle flags that are off remotely.

For the use case of tweaking it in `./test/integration`, I'm not entirely sure how much is useful, because to have it truly simulate a legacy `struct raft` we should have it run the same code paths/logic, which are typically not available in the current version of the code.

Perhaps to test that behavior we can rather feed `struct raft` with a manually crafted `struct raft_message`. For example a `struct raft_append_entries_result` with flags set to 0, and then check that the node sends out a single-chunked `struct raft_install_snapshot` instead of a multi-chunked one.